js/memory.js

```diff
@@ -1,9 +1,42 @@
 class MemoryGame {
   constructor(cards){
     this.cards = cards;
+    this.pickedCards = [];
+    this.pairsClicked = 0;
+    this.pairsGuessed = 0;
+    
     // add the rest of the class properties here
   }
-  shuffleCards() {}
-  checkIfPair(card1, card2) {}
-  isFinished() {}
-}
\ No newline at end of file
+  shuffleCards(array) {
+    let card, shuffle, i; 	    
+    
+    if (!this.cards) {
+        return undefined;
+      }
+      for (i = this.cards.length - 1; i > 0; i--) {
+        card = Math.floor(Math.random() * (i + 1));
+        shuffle = this.cards[i];
```

Comment:
Is "shuffle" an apt name for what this variable represents? Same question for the "card" variable.